app/models/concerns/host_common.rb

```diff
@@ -19,6 +19,8 @@ module HostCommon
     belongs_to :puppet_ca_proxy, :class_name => "SmartProxy"
     belongs_to :realm,           :counter_cache => counter_cache
     belongs_to :compute_profile
+    validates :puppet_ca_proxy, :proxy_features => { :feature => "Puppet CA", :message => _("does not have the Puppet CA feature") }
```

Comment:
use `N_()` so the English is always stored in the class for translation later, since there's no locale (or an undetermined locale at least) when the class is evaluated.